tests/common/plugins/pdu_controller/__init__.py

```diff
@@ -1,9 +1,20 @@
 import logging
 
 import pytest
+import os
+import sys
 from .pdu_manager import pdu_manager_factory
 from tests.common.utilities import get_host_visible_vars, get_sup_node_or_random_node
 
+SELF_DIR = os.path.dirname(os.path.abspath(__file__))
+BASE_DIR = os.path.realpath(os.path.join(SELF_DIR, ".."))
+if BASE_DIR not in sys.path:
+    sys.path.append(BASE_DIR)
+ANSIBLE_DIR = os.path.realpath(os.path.join(SELF_DIR, "../../../../ansible"))
+if ANSIBLE_DIR not in sys.path:
+    sys.path.append(ANSIBLE_DIR)
+
+from devutil.inv_helpers import HostManager            # noqa E402
```

Comment:
I am a little bit concerned with adding this new dependency. Purpose of HostManager is to find some hosts from inventory, right?

Is there other way to do this without this new dependency? Can we take advantage of the InventoryManager in duthosts.host.options["inventory_manager"]?

Or enhance the design of conn_graph_facts for VS setup a little bit?

Reply:
The code in branch `if hostname not in device_pdu_links or hostname not in device_pdu_info:` is unnecessary here. Because, we use `conn_graph_facts` to get pdu links and pdu info first. `if the hostname not in device_pdu_links or hostname not in device_pdu_info` here means the host doesn't not exist in the csv file, so we don't have to get info from inventory. Am I right?

And @Xichen96, do you agree with me?

Reply:
Also, if we try to get pdu info form inventory, unfortunately, for most DUTs, we will get `None` because there is no key `pdu_host` under most of the hosts. And although we can get the pdu hosts list of a DUT from inventory, we can not get the hwsku and os of pdu host from inventory. Inventory doesn't supply such information. So I think the code in this if branch is redundant and remove them.